File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    
    numer = numer1 * denom2 + numer2 * denom1
    denom = denom1 * denom2
    
    #n = max(n for n in range(1, min(numer, denom)) if numer%n==0 and denom %n==0)
    
    n = 2;
    while (n <= min(numer, denom)):
        if numer%n==0 and denom%n==0:
            numer /= n
            denom /= n
        else: n += 1
    
    return [numer, denom]

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int numer = numer1 * denom2 + numer2 * denom1;
        int denum = denom1 * denom2;
        
        int i = 2;
        while (i <= Math.min(numer, denum)) {
            if (numer % i == 0 && denum % i == 0) {
                numer /= i;
                denum /= i;
            } else i++;
        }
        
        return new int[] {numer, denum};
    }
}